README.md

```diff
@@ -1,104 +1,151 @@
 
-**NOTE:** It might take some time for the build to finish as it depends on the number of jobs in the queue. If it's taking long, [contact us](#get-in-touch).
+How do I package that application into a container that's updated automatically every time I push changes? What about security and updates, how do I automate that each time I push any changes?
 
-The entire flow can be summarized as below
+That's where CCCP comes in. 
 
-![Container Pipeline Diagram](docs/diagrams/architecture.png)
+CCCP will:
+
+  - Scan the image for updates, fixes, capabilities and push it to a public registry (by default, http://registry.centos.org)
+  - Automatically rebuild when an RPM package is updated in the repository or base image (`FROM` in Dockerfile) is updated
+
+## How do I host my application?
+
+Similar to projects such as [Homebrew](https://github.com/Homebrew/homebrew-core) it's as easy as opening up a pull request.
+
+A developer wishing to host their container image will open up a pull request to the [CentOS Container Index](https://github.com/CentOS/container-index). 
+
+Once the pull request is merged, CCCP:
 
-1. **Project onboarding**
+  1. Links the Dockerfile
+  2. Builds the image
+  3. Scans / analyzes it
+  4. Pushes to [registry.centos.org](https://registry.centos.org)
+  5. Notifies the developer (email)
 
-    Refer the [Container Index](https://github.com/CentOS/container-index).
-    
-2. **Jenkins based tracking**
+Once a project is in the [CentOS Container Index](https://github.com/CentOS/container-index), the CentOS Container Pipeline Service service automatically tracks the project's Git repo and branch and rebuilds it every time there is a future change.
 
-    Tracks the developer's Git repository + branch for any change and triggers a new build on OpenShift when a change is pushed. Along with developer's repo, an update in base image or any of the RPMs which are a part of the image, also trigger a fresh build.
+## How everything works
 
-3. **Build the image**
+1. **Project onboard / the main "index"**
 
-    Build the container image using the targetfile (Dockerfile) and push it to OpenShift's internal registry. Result: image tagged as `:test` pushed to internal registry.
+First off, the pipeline points to an index. For the CentOS community and in our example, this refers to the: [CentOS Container Index](https://github.com/CentOS/container-index). 
+
+2. **Jenkins and OpenShift tracking**
+
+Jenkins is utilized in order to track each application's Git repository as well as branch for any changes. This triggers a new build on **OpenShift** when a change is pushed.
+
+Changes to the application's repository, update to the base image or any RPMs that are part of the image will trigger a new build
+
+3. **Building the image**
+
+The container image is built within OpenShift and then pushed to OpenShift's internal registry. This results in an image tagged `image:test`.
```

Comment:
Originally, we used to push the image to Openshift internal registry. However, due to multiple issues, faced multiple times, for bringing up openshift internal registry, I believe we started to push to the public registry, with a temporary tag generated by hashing stuff like datetime